visualiser/visualiser.py

```diff
@@ -78,8 +78,9 @@ def make_frames(cls):
     @classmethod
     def write_gif(cls, name="out.gif", delay=3):
         images = []
-        imgs = glob.glob("frames/*.png")
-        for filename in sorted(imgs, key=lambda fn: int(fn.split("_")[1].split(".")[0])):
+        sorted_images = sorted(glob.glob("frames/*.png"))
```

Comment:
We need to do integer sort i.e 
` ['frames/temp_0.png', 'frames/temp_1.png', 'frames/temp_10.png', 'frames/temp_11.png', 'frames/temp_12.png', 'frames/temp_13.png', 'frames/temp_14.png', 'frames/temp_15.png', 'frames/temp_16.png', 'frames/temp_17.png', 'frames/temp_18.png', 'frames/temp_19.png', 'frames/temp_2.png', 'frames/temp_20.png', 'frames/temp_21.png', 'frames/temp_22.png', 'frames/temp_23.png', 'frames/temp_24.png', 'frames/temp_3.png', 'frames/temp_4.png', 'frames/temp_5.png', 'frames/temp_6.png', 'frames/temp_7.png', 'frames/temp_8.png', 'frames/temp_9.png']`

But the actual sort order should be:
```
frames/temp_0.png
frames/temp_1.png
frames/temp_2.png
frames/temp_3.png
frames/temp_4.png
frames/temp_5.png
frames/temp_6.png
frames/temp_7.png
frames/temp_8.png
frames/temp_9.png
frames/temp_10.png
frames/temp_11.png
frames/temp_12.png
frames/temp_13.png
frames/temp_14.png
frames/temp_15.png
frames/temp_16.png
frames/temp_17.png
frames/temp_18.png
frames/temp_19.png
frames/temp_20.png
frames/temp_21.png
frames/temp_22.png
frames/temp_23.png
frames/temp_24.png
```

Reply:
yes! key was important here
```
        # sort frames images in ascending order
        sorted_images = sorted(
            glob.glob("frames/*.png"),
            key=lambda fn: int(fn.split("_")[1].split(".")[0])
        )
```

Reply:
Can you rename fn to file_name or sth meaningful and maybe add an example as a comment of what the lambda function is doing?

Reply:
okay I will look into it.